fakedata.cabal

```diff
@@ -677,7 +677,8 @@ library
   hs-source-dirs:
       src
   build-depends:
-      attoparsec
+      aeson >= 2.0
```

Comment:
Is it possible to support both aesons ? Unfortunately aeson 1.x is still present in stackage nightly and I believe it would take some time for the ecosytem to catch up with 2.0.

Reply:
Basically the idea is that you run the `run.sh` script: https://github.com/fakedata-haskell/fakedata/blob/master/scripts/run.sh

For targetting a particular module, you have to change the `currentOne` type to point to a particular module: https://github.com/fakedata-haskell/fakedata/blob/0e9ab13adae39654e85981de3a329b5fd9f1f424/scripts/FakeModuleInfo.hs#L2361

That being said, I would say that the scripts are not perfect and I usually manually do the changes to the generated code to handle edge cases etc. So in summary, the scripts are a good starting point for a new module, but not so good for the existing modules...

Reply:
Ok, then I'm unlikely to be able to support `aeson-1` and `aeson-2` as that will be significant work to do this on all modules. If it were me, I'd just cut a new release that supports `aeson-2`. If people need `aeson-1`, they can always use an older version.

Reply:
I would like to merge this MR and add support for aeson-1 myself when time allows me. Can you make sure that the test suite passes ? Right now, I seem to be getting a compile error:

```
/home/sibi/github/fakedata/test/AddressSpec.hs:27:20: error:
    * Couldn't match type `Data.Aeson.Key.Key' with `Text'
      Expected type: IO Text
        Actual type: IO Data.Aeson.Key.Key
    * In the expression: pure t
      In the third argument of `resolveUnresolved', namely
        `(\ s t -> pure t)'
      In a stmt of a 'do' block:
        txt <- resolveUnresolved
                 defaultFakerSettings (pure $ pure $ "32-????-####")
                 (\ s t -> pure t)
   |
27 |           (\s t -> pure t)
   |                    ^^^^^^

/home/sibi/github/fakedata/test/AddressSpec.hs:34:20: error:
    * Couldn't match type `Data.Aeson.Key.Key' with `Text'
      Expected type: IO Text
        Actual type: IO Data.Aeson.Key.Key
    * In the expression: pure t
      In the third argument of `resolveUnresolved', namely
        `(\ s t -> pure t)'
      In a stmt of a 'do' block:
        txt <- resolveUnresolved
                 defaultFakerSettings (pure $ pure $ "abjakf-324jak")
                 (\ s t -> pure t)
   |
34 |           (\s t -> pure t)
   |                    ^^^^^^

Progress 1/2
```

Reply:
Thanks, I didn't think to update the tests :facepalm: I'll get those fixed up now

Reply:
Ok, tests are passing again.